mixer/cmd/build.go

```diff
@@ -280,6 +280,14 @@ var buildFormatOldCmd = &cobra.Command{
 		oldFormat := b.State.Mix.Format
 		b.State.Mix.Format = buildFlags.newFormat
 
+		// Delete index manifest when passing format 28
+		if oldFormat <= "28" && b.State.Mix.Format > "28" {
```

Comment:
This passed by me unnoticed at first, but you are comparing strings, not ints. You need to convert or this won't work as intended.